libs/brain/input-otp/src/lib/brn-input-otp.component.ts

```diff
@@ -0,0 +1,127 @@
+import { BooleanInput, NumberInput } from '@angular/cdk/coercion';
+import {
+	booleanAttribute,
+	Component,
+	computed,
+	forwardRef,
+	input,
+	model,
+	numberAttribute,
+	signal,
+} from '@angular/core';
+import { ControlValueAccessor, FormsModule, NG_VALUE_ACCESSOR } from '@angular/forms';
+import { ChangeFn, TouchFn } from '@spartan-ng/brain/forms';
+import { provideBrnInputOtp } from './brn-input-otp.token';
+
+export const BRN_INPUT_OTP_VALUE_ACCESSOR = {
+	provide: NG_VALUE_ACCESSOR,
+	useExisting: forwardRef(() => BrnInputOtpComponent),
+	multi: true,
+};
+
+export type InputMode = 'text' | 'tel' | 'url' | 'email' | 'numeric' | 'decimal' | 'search';
+
+@Component({
+	selector: 'brn-input-otp',
+	standalone: true,
+	imports: [FormsModule],
+	template: `
+		<ng-content />
+		<div style="position: absolute; inset: 0; pointer-events: none;">
+			<input
+				[class]="inputClass()"
+				autocomplete="one-time-code"
+				data-slot="input-otp"
+				style="position: absolute; inset: 0; width: 100%; height: 100%; display: flex; textAlign: left; opacity: 1; color: transparent; pointerEvents: all; background: transparent; caret-color: transparent; border: 0px solid transparent; outline: transparent solid 0px; box-shadow: none; line-height: 1; letter-spacing: -0.5em; font-family: monospace; font-variant-numeric: tabular-nums;"
+				[disabled]="state().disabled()"
+				[inputMode]="inputMode()"
+				[ngModel]="value()"
+				(input)="onInputChange($event)"
+				(focus)="focused.set(true)"
+				(blur)="focused.set(false)"
```

Comment:
We should also add a `(paste)` handler in case users copy the OTP from a 2FA app, email, or wherever. We can take this up in a separate PR though, or if I missed it somewhere in this PR then please ignore this.

Reply:
Copy and paste already works, but we can add `(paste)` event and handle it a bit more.
What would be the use case for `(paste)`? Like removing whitespaces or keeping only the last part of a long text matching the maxLength?

Reply:
Good idea. I added `transformPaste` input to allow the for custom transformation functions

`transformPaste: (pastedText: string, maxLength: number) => pastedText.replaceAll('-', '').slice(0, maxLength)`